src/generator.rs

```diff
@@ -0,0 +1,124 @@
+    type Item = (Robot, Vec<Command>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let rng = &mut self.prng;
+
+        let robot = Robot {
+            position: Point {
+                x: rng.gen_range(0, self.upper_right.x),
+                y: rng.gen_range(0, self.upper_right.y),
+            },
+            facing: rng.gen(),
+        };
+
+        let ncmds = rng.gen_range(1, 100);
+        let commands = rng.sample_iter(Standard).take(ncmds).collect();
+
+        Some((robot, commands))
+    }
+}
+
+// Randomly generated custom types
+
+impl Distribution<Orientation> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Orientation {
+        match rng.gen_range(0, 4) {
+            0 => Orientation::North,
+            1 => Orientation::West,
+            2 => Orientation::South,
+            _ => Orientation::East,
+        }
+    }
+}
+
+impl Distribution<Command> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Command {
+        match rng.gen_range(0, 3) {
+            0 => Command::Left,
+            1 => Command::Right,
+            _ => Command::Forward,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::geo::orientation::Orientation::{North as N, South as S};
+    use crate::robot::Command::{Forward as F, Left as L, Right as R};
+
+    #[test]
+    fn creates_mission() {
+        let generator = Generator::new(12345);
+        let expected = Point { x: 17, y: 43 };
+
+        assert_eq!(generator.upper_right, expected);
+    }
+
+    #[test]
+    fn generates_three_robots() {
+        let generator = Generator::new(12345);
+        let expected = vec![
+            (
+                Robot {
+                    position: Point { x: 16, y: 39 },
+                    facing: S,
+                },
+                vec![
+                    R, L, R, L, R, L, R, L, F, F, R, L, R, R, F, R, L, F, L, R, L, F, R, F, R, R,
+                    F, L, F, L, L, F, L, F, F, L, F, R, L, L, F, R, R, R, L, F, R, L, R, R, R, L,
+                    F, R, L, L, F, R,
+                ],
+            ),
+            (
+                Robot {
+                    position: Point { x: 12, y: 32 },
+                    facing: S,
+                },
+                vec![
+                    L, R, F, F, F, R, L, R, L, L, R, L, R, L, F, F, R, F, L, F, L, F, L, R, R, F,
+                    F, F, R, F, F, F, L, F, R, L, F, F, L, L, F, R, L, R, F, F, F, F, R, R, R, R,
+                    R, F, F, L, R, R, L, R, F, R, L, R, F, R, F, L, R, L, L, R, L, R,
+                ],
+            ),
+            (
+                Robot {
+                    position: Point { x: 7, y: 42 },
+                    facing: N,
+                },
+                vec![
+                    L, L, R, L, L, R, R, R, R, L, F, F, F, R, L, R, L, F, L, F, L, R, F, R, L, R,
+                    L, L, R, F, R, R, R, R,
+                ],
+            ),
+        ];
```

Comment:
Another library suggestion - assume you just got these by running the code & copying & pasting in.  We've been using [`insta`](https://docs.rs/insta/0.15.0/insta/) for this sort of thing - it is pretty nice.

Reply:
Oh sweet, I was wondering whether there's tooling for this, thanks!